kupyna/src/compress512.rs

```diff
@@ -0,0 +1,177 @@
+use crate::tables::{MDS_MATRIX, SBOXES};
+
+pub(crate) const COLS: usize = 8;
+const ROUNDS: u64 = 10;
+
+const BITS_IN_BYTE: u8 = 8;
+const REDUCTION_POLYNOMIAL: u16 = 0x011d;
+
+type Matrix = [[u8; 8]; 8];
+
+pub(crate) fn compress(prev_vector: &mut [u64; COLS], message_block: &[u8; 64]) {
+    let mut prev_vector_u8 = [0u8; 64];
+    for (i, &value) in prev_vector.iter().enumerate() {
+        let bytes = value.to_be_bytes();
+        prev_vector_u8[i * 8..(i + 1) * 8].copy_from_slice(&bytes);
```

Comment:
Usually it's better to transform message block into `[u64; 8]`, than transforming the state into bytes.

Try to look into re-implementing `t_xor_l` and `t_plus_l` in terms of `[u64; 8]`? Do not forget to measure performance while doing it.